quiver_engine/imagenet_utils.py

```diff
@@ -8,28 +8,36 @@
 
 from keras.utils.data_utils import get_file
 from keras import backend as K
+import numpy as np
 
 CLASS_INDEX = None
 CLASS_INDEX_PATH = 'https://s3.amazonaws.com/deep-learning-models/image-models/imagenet_class_index.json'
+EPS = np.finfo(np.float32).eps
 
 
-def preprocess_input(x, dim_ordering='default'):
+def preprocess_input(x, dim_ordering='default', mean=None, std=None):
     if dim_ordering == 'default':
         dim_ordering = K.image_dim_ordering()
     assert dim_ordering in {'tf', 'th'}
 
-    if dim_ordering == 'th':
-        x[:, 0, :, :] -= 103.939
-        x[:, 1, :, :] -= 116.779
-        x[:, 2, :, :] -= 123.68
-        # 'RGB'->'BGR'
-        x = x[:, ::-1, :, :]
-    else:
-        x[:, :, :, 0] -= 103.939
-        x[:, :, :, 1] -= 116.779
-        x[:, :, :, 2] -= 123.68
-        # 'RGB'->'BGR'
-        x = x[:, :, :, ::-1]
+    if mean is not None:
+        x = x - np.array(mean, dtype='float32')
+    if std is not None:
+        x = x / (np.array(std, dtype='float32') + EPS)
```

Comment:
Trying to understand why you need the machine epsilon here - what's an example where not adding it will mess things up?

Reply:
I took the inspiration from Keras https://github.com/fchollet/keras/blob/master/keras/preprocessing/image.py#L501, but now I see that it's not needed here.